runners/direct-java/src/test/java/org/apache/beam/runners/direct/WriteWithShardingFactoryTest.java

```diff
@@ -163,74 +163,81 @@ public void withNoShardingSpecifiedReturnsNewTransform() {
   }
 
   @Test
-  public void keyBasedOnCountFnWithNoElements() throws Exception {
+  @Category(ValidatesRunner.class)
```

Comment:
The category here should not be necessary. This is  a specialized transform just for the DirectRunner so the runner should be hardcoded.